CMakeLists.txt

```diff
@@ -36,6 +37,24 @@ if (MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -wd4125")
 endif()
 
+# Enable ccache
+# This setup is sufficient for Ninja and Makefiles, but not Xcode
+if (ONNX_MLIR_CCACHE_BUILD)
+  find_program(CCACHE_PROGRAM ccache)
+  if (CCACHE_PROGRAM)
+    if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+      # RULE_LAUNCH_COMPILE is more powerful than C_COMPILER_LAUNCHER (ccache
+      # supports additional compilers) but has issues with cmd and MSVC tools
+      set(CMAKE_C_COMPILER_LAUNCHER ${CCACHE_PROGRAM})
+      set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_PROGRAM})     
+    else()
+      set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_PROGRAM})
+    endif()
+  else()
+    message(FATAL_ERROR "Unable to find the program ccache. Set ONNX_MLIR_CCACHE_BUILD to OFF.")
```

Comment:
I wonder whether it's better to just issue a warning and continue? Either way is fine I guess. No strong preference to either.

Reply:
I copied the LLVM behavior here for consistency. If you would prefer it changed I can, but my concern would be that the CMake generate for ONNX MLIR is a pretty long chunk of text, and it's easy to miss warnings in there.